### ISSUE TITLE
Multiple Auth setups at AWSMobileClient

### ISSUE THREAD

**Issue by VinnyScorsatto** [aws-amplify/aws-sdk-ios#2388]
Hello, I am implementing Cognito Auth to my Swift App. 

I have initialised my instance of AWSMobileClient by constructing it with AWSMobileClient(configuration). My configuration looks like this:

{
  "CognitoUserPool": {
    "Default": {
      "PoolId": "XX-XXXX-X_abcd1234",
      "AppClientId": "XXXXXXXX",
      "AppClientSecret": "XXXXXXXXX",
      "Region": "XX-XXXX-X"
    }
  },
  "Auth": {
    "Default": {
      "authenticationFlowType": "CUSTOM_AUTH"
    }
  }
}

My issue is after using CUSTOM_AUTH, I need to also use USER_PASSWORD_AUTH, but according to the docs, it is possible to run do AWSMobileClient(configuration) only once, is there something that I am missing here? Is there a way to configure an alternative Auth or reload the configuration?

"aws-amplify/aws-sdk-ios" "2.13.0" - Carthage
iOS 10.0 or Higher

**Comment by wooj2** [comment]
Hey @VinnyScorsatto,

Thanks for reaching out.  Just trying to better understand your use case--  are you attempting to support a CUSTOM_AUTH and USER_SRP_AUTH in the same application?  If so, I don't believe this is currently a supported use case for the AWSMobileClient.

**Comment by VinnyScorsatto** [comment]
Hey @wooj2 , thanks for reaching back,

Yes, we actually do it on our web app by starting with CUSTOM_AUTH and them after we call Amplify.configure to turn to USER_PASSWORD_AUTH. Thanks

**Comment by wooj2** [comment]
Hey @VinnyScorsatto ,

I checked with a couple of teammates and your understanding is correct - the AWSMobileClient does not support multiple authenticationFlowTypes.  Sorry about the inconvenience.

**Comment by VinnyScorsatto** [comment]
Hi @wooj2 , thanks a lot

**Comment by hossamghareeb** [comment]
@VinnyScorsatto There is a hacky way I used to solve this problem. I don't add `CUSTOM_AUTH` in json but I do the following when I want to perform custom auth:
```Swift
let pool = AWSCognitoIdentityUserPool.default()
pool.isCustomAuth = true
```

**Comment by VinnyScorsatto** [comment]
Hi @hossamghareeb , thanks for your message.

Your suggestion is to initialise the AWSMobileClient with this config

{
"CognitoUserPool": {
"Default": {
"PoolId": "XX-XXXX-X_abcd1234",
"AppClientId": "XXXXXXXX",
"AppClientSecret": "XXXXXXXXX",
"Region": "XX-XXXX-X"
}
},
"Auth": {
"Default": {
"authenticationFlowType": "USER_PASSWORD_AUTH"
}
}
}

and then set pool.isCustomAuth = true, do the first part of the AUTH using CUSTOM_AUTH and then set it to false, so it would get the configuration USER_PASSWORD_AUTH from the awsconfiguration.json? 

Also, I am using AWSMobileClient, so is it ok to use AWSCognitoIdentityUserPool combined to it?

Thanks a lot for your suggestion

**Comment by hossamghareeb** [comment]
Yes true. We also use `AWSMobileClient` and we have 2 ways of authentication, one with `USER_SRP_AUTH` and one with `CUSTOM_AUTH`. We don't specify anything regarding `authenticationFlowType` in the configuration.

**Comment by VinnyScorsatto** [comment]
Hey @hossamghareeb , that is awesome, would you mind sharing the code on how you initialise your instance of AWSMobileClient?

Also, could you share the structure of your awsconfiguration.json?

Thanks a lot for your support

**Comment by VinnyScorsatto** [comment]
@hossamghareeb , just to add, I have checked again the docs and I believe I understand how your app is working.. USER_SRP_AUTH is the default AUTH for mobile and pool.isCustomAuth = true really works, but in my case I have 2 non-defaults. Also it seems that for some reason the SDK is ignoring USER_PASSWORD_AUTH (I have opened an issue for that). So I am still trying to figure out what to do, but to be honest I am running out of options. Thanks
